### PR TITLE
Run build on JDK 21

### LINKED ISSUE #6591: Run CI on JDK 21
## Expected Behavior
CI runs for all LTS Java versions.

## Current Behavior
We currently run on Java 8, 11 and 17.

## Context
Keep CI current.

This is a placeholder issue raised while I remember it - blocked until we update to Kotlin 1.9.20 and Gradle 8.5.

### PR DESCRIPTION
**Pull request by 3flex**
Closes #6591

### CODE REVIEW

**Comment by Anksmahajan** [comment]
Gradle 8.5 is out now. Any plans on taking this PR forward?

**Comment by 3flex** [comment]
Yes

File: .github/workflows/pre-merge.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ ubuntu-latest, macos-latest, windows-latest ]
-        jdk: [ 8, 11, 17 ]
+        jdk: [ 8, 11, 17, 21 ]
```
Comment (BraisGabin) [inline .github/workflows/pre-merge.yaml:25]: Could we remove any of the others? We already have a lot of tasks I would prefer to don't add 3 more.

Comment (3flex) [reply .github/workflows/pre-merge.yaml:25]: Agree but can we look at that separately?

**Pull request merged**